### PR TITLE
Binary transform for br and br_if.

### LINKED ISSUE #8614: wasm: implement br and br_if
Implement br and br_if in wasm/BinaryTransform.cpp

### PR DESCRIPTION
**Pull request by chriseth**
Closes #8614

### CODE REVIEW

**Comment by axic** [comment]
Can you also fix the `break` to `br` in the text representation?

**Comment by chriseth** [comment]
@axic did you even read my code??? https://github.com/ethereum/solidity/pull/8918/files#diff-d08f85d3f7988665655f040b9ffccf82R126 ;)

**Comment by axic** [comment]
No, I haven't. But had that change lying around was just lazy to push a sad single-line PR.

File: libyul/backends/wasm/BinaryTransform.cpp [hunk 1]
```diff
@@ -581,6 +586,15 @@ bytes BinaryTransform::visitReversed(vector<Expression> const& _expressions)
 	return result;
 }
 
+bytes BinaryTransform::encodeLabelIdx(string const& _label) const
+{
+	yulAssert(!_label.empty(), "Empty label.");
+	auto it = find(m_labels.rbegin(), m_labels.rend(), _label);
+	yulAssert(it != m_labels.rend(), "Label not found.");
+	size_t depth = it - m_labels.rbegin();
```
Comment (ekpyron) [inline libyul/backends/wasm/BinaryTransform.cpp]:
```suggestion
	size_t depth = 0;
	for(auto const& label: m_labels | reversed)
	{
	    if (label == _label) return lebEncode(depth);
	    ++depth;
	}
	yulAssert(false, "Label not found.");
```
maybe?

**Review by ekpyron — DISMISSED** [review]
I'm not too familiar with the wasm backend, but this makes sense and looks fine to me!
We don't have a proper way of testing the wasm binary, do we?

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `c513cc411db0bd53f946af546e3d1d85b259f293`:
```
Coding style error:
 libyul/backends/wasm/BinaryTransform.cpp:593: for(auto const& label: m_labels | reversed)
 libyul/backends/wasm/BinaryTransform.cpp:595: if (label == _label) return lebEncode(depth);
 libyul/backends/wasm/BinaryTransform.cpp:596: ++depth;

```
Please check that your changes are working as intended.

File: libyul/backends/wasm/BinaryTransform.cpp [hunk 2]
```diff
@@ -581,6 +586,19 @@ bytes BinaryTransform::visitReversed(vector<Expression> const& _expressions)
 	return result;
 }
 
+bytes BinaryTransform::encodeLabelIdx(string const& _label) const
+{
+	yulAssert(!_label.empty(), "Empty label.");
+	size_t depth = 0;
+	for(auto const& label: m_labels | reversed)
```
Comment (stackenbotten) [inline libyul/backends/wasm/BinaryTransform.cpp]: Coding style error

File: libyul/backends/wasm/BinaryTransform.cpp [hunk 3]
```diff
@@ -581,6 +586,19 @@ bytes BinaryTransform::visitReversed(vector<Expression> const& _expressions)
...
+	{
+	    if (label == _label) return lebEncode(depth);
```
Comment (stackenbotten) [inline libyul/backends/wasm/BinaryTransform.cpp]: Coding style error

File: libyul/backends/wasm/BinaryTransform.cpp [hunk 4]
```diff
@@ -581,6 +586,19 @@ bytes BinaryTransform::visitReversed(vector<Expression> const& _expressions)
...
+	    ++depth;
```
Comment (stackenbotten) [inline libyul/backends/wasm/BinaryTransform.cpp]: Coding style error

**Pull request merged**